pkg/webconsoleurl/url_lookup.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package webconsoleurl
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appv1a1 "github.com/vmware-tanzu/vm-operator/external/appplatform/api/v1alpha1"
+	pkgcfg "github.com/vmware-tanzu/vm-operator/pkg/config"
+)
+
+const (
+	SupervisorServiceObjName      = "supervisor-env-props"
+	SupervisorServiceObjNamespace = "vmware-system-supervisor-services"
+
+	ProxyAddrServiceName      = "kube-apiserver-lb-svc"
+	ProxyAddrServiceNamespace = "kube-system"
+)
+
+// ProxyAddress first attempts to get the proxy address through the API Server DNS Names.
+// If that is unset, though, fall back to using the virtual IP.
+func ProxyAddress(ctx context.Context, r client.Client) (string, error) {
+	if !pkgcfg.FromContext(ctx).Features.SimplifiedEnablement {
+		return ProxyAddressFromVirtualIP(ctx, r)
+	}
+
+	// Attempt to use the API Server DNS Names to get the proxy address.
+	proxyAddress, err := ProxyServiceDNSName(ctx, r)
+	if err != nil {
+		return "", fmt.Errorf("failed to get proxy service URL: %w", err)
+	}
+
+	// If no API Server DNS Name exists, fall back to using the virtual IP.
+	if proxyAddress == "" {
+		return ProxyAddressFromVirtualIP(ctx, r)
+	}
+
+	return proxyAddress, nil
+}
+
+// ProxyServiceDNSName retrieves the first API server DNS name using the provided client by
+// querying the appplatform CRD, if one exists.
+func ProxyServiceDNSName(ctx context.Context, r client.Client) (string, error) {
```

Comment:
I don't think either `ProxyService{DNSName,FromVirtualIP}` needs to be public

Reply:
Changed to private.